### PR TITLE
Add page description and taxonomies to lunr search

### LINKED ISSUE #389: Description not used for search
It looks like the description text from the frontmatter is not used by Lunr search.

### PR DESCRIPTION
**Pull request by narrenfrei**
Add page description and taxonomies to lunr.js search index
changes in offline-search-index.json and offline-search.js

Fixes #389

### CODE REVIEW

File: assets/json/offline-search-index.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {{- $.Scratch.Add "offline-search-index" slice -}}
 {{- range where .Site.AllPages ".Params.exclude_search" "!=" true -}}
 {{/* We have to apply `htmlUnescape` again after `truncate` because `truncate` applies `html.EscapeString` if the argument is not HTML. */}}
-{{- $.Scratch.Add "offline-search-index" (dict "title" .Title "ref" .RelPermalink "body" (.Plain | htmlUnescape) "excerpt" (.Plain | htmlUnescape | truncate (.Site.Params.offlineSearchSummaryLength | default 70) | htmlUnescape)) -}}
+{{/* Indvidual taxonomies can be added in the next line by add '"taxonomy-name" (.Params.taxonomy-name | default "")' to the dict (as seen for categories and tags). */}}
+{{- $.Scratch.Add "offline-search-index" (dict "ref" .RelPermalink "title" .Title "categories" (.Params.categories | default "") "tags" (.Params.tags | default "") "description" (.Description | default "") "body" (.Plain | htmlUnescape) "excerpt" ((.Description | default .Plain) | htmlUnescape | truncate (.Site.Params.offlineSearchSummaryLength | default 200) | htmlUnescape)) -}}
```
Comment (gwatts) [inline assets/json/offline-search-index.json:5]: Increasing the excerpt length to 200 makes the result box larger than i'd like.. 70 worked much better for me and seems sufficient (at least for my docs use case 😄 )

**Comment by gwatts** [comment]
Thanks for tackling this!  Looks great to me, other than the results box being much wider due to the increased excerpt size - Haven't tested adding taxonomies though

**Comment by narrenfrei** [comment]
> 
> 
> Thanks for tackling this! Looks great to me, other than the results box being much wider due to the increased excerpt size - Haven't tested adding taxonomies though

Thanks for your comment and sorry - it wasn't planned to increase the default value for excerpt truncation. I'd made copy and paste from my project, where I've tested it.

**Comment by LisaFC** [comment]
This looks great (and seems to work, though also haven't tested it with taxonomies).

**Pull request merged**